### ISSUE TITLE
Access-Control-Allow-Origin issue on login

### ISSUE THREAD

**Issue by cforce** [Azure-Samples/azure-search-openai-demo#2102]
On initial request https://example.com i get below CORS issue i can't explain. The background image is loaded and login redirect did not happen. Then when i reload the page again the redirect works as expected

Access to fetch at 'https://login.microsoftonline.com/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/oauth2/v2.0/authorize?response_type=code+id_token&redirect_uri=https%3A%2F%example.com%2F.auth%2Flogin%2Faad%2Fcallback&client_id=XXXXXXXXXXXXXXXXXXXXXXXX&scope=api%3A%2F%2FXXXXXXXXXXXXXXXXXXXXXXXX.default+openid+profile+email+offline_access&response_mode=form_post&nonce=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&state=redir%3D%252Fauth_setup' (redirected from 'https://example.com/auth_setup') from origin 'https://example.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled

**Comment by pamelafox** [comment]
I think @mattgotteiner has seen CORS issues in the past. Are you using user login on App Service with built-in auth? Or are you on Azure Container Apps?

**Comment by cforce** [comment]
It is fixed when I set ALLOW_ORGIN to my domain which shall not needed according to docs- is it?

**Comment by pamelafox** [comment]
I'm making a few CORS related fixes today. Once made, you should see this in your allowed CORS origins:
![Screenshot 2024-12-02 at 12 37 49 PM](https://github.com/user-attachments/assets/99254f37-4b4e-44ff-8d4b-a53e36760c11)

And this for the ALLOWED_ORIGIN environment variable, which is used by the app's CORS plugin:
"https://portal.azure.com;https://ms.portal.azure.com;https://login.microsoftonline.com"

That should be all that's needed when using login.

**Comment by pamelafox** [comment]
Closing as a duplicate of #1780 — same root cause (the frontend does `fetch('/auth_setup')`, Container Apps Easy Auth / App Service Auth returns a 302 to `login.microsoftonline.com`, and the browser refuses to follow the cross-origin redirect on a `fetch`). The trigger differs only in when it happens (initial load vs. after logout), but the code path and fix are the same. Tracking further discussion and the fix in #1780.

Reproduced today on a fresh Container Apps + Entra deploy — confirms this is still current.
